x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals_histogram_panel/signals_histogram/helpers.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HistogramData, SignalsAggregation, SignalsBucket, SignalsGroupBucket } from '../types';
+import { SignalSearchResponse } from '../../../../../containers/detection_engine/signals/types';
+import * as i18n from '../translations';
+
+export const formatSignalsData = (
+  signalsData: SignalSearchResponse<{}, SignalsAggregation> | null
+) => {
+  const groupBuckets: SignalsGroupBucket[] =
+    signalsData?.aggregations?.signalsByGrouping?.buckets ?? [];
+  let result: HistogramData[] = [];
+  groupBuckets.forEach(({ key: group, signals }) => {
+    const signalsBucket: SignalsBucket[] = signals?.buckets ?? [];
```

Comment:
Are your types correct?

```
export interface SignalsGroupBucket {
  key: string;
  signals: {
    buckets: SignalsBucket[];
  };
}
```

You have that which indicates that both signals and buckets can't be null/undefined but you're using the `?` operator in both cases. Should you change your types to match what you're seeing during runtime or is it the other way around?

Also, nit: you can remove the `let result` but using a reduce here no?

Reply:
Nice catch! Types are correct, just unnecessary use of `?`, so I'll clean that and the `let` up. 👍